src/index.ts

```diff
@@ -0,0 +1,112 @@
+import { spawn } from 'child_process';
+import { existsSync, mkdirSync, copyFileSync, chmodSync } from 'fs';
+import { tmpdir } from 'os';
+import { resolve, join, basename } from 'path';
+import defaultLogger, { Logger } from './utils/log';
+import { version } from '../package.json';
+
+enum CloudEnvironment {
+  AZURE_FUNCTION = 'Azure Function',
+  GOOGLE_CLOUD_RUN_FUNCTION_1ST_GEN = 'Google Cloud Run Function 1st gen',
+  GOOGLE_CLOUD_RUN_FUNCTION_2ND_GEN = 'Google Cloud Run Function 2nd gen',
+  UNKNOWN = 'Unknown',
+}
+
+function getEnvironment(): CloudEnvironment {
+  if (
+    process.env.FUNCTIONS_EXTENSION_VERSION !== undefined &&
+    process.env.FUNCTIONS_WORKER_RUNTIME !== undefined
+  ) {
+    return CloudEnvironment.AZURE_FUNCTION;
+  }
+
+  if (
+    process.env.FUNCTION_NAME !== undefined &&
+    process.env.GCP_PROJECT !== undefined
+  ) {
+    return CloudEnvironment.GOOGLE_CLOUD_RUN_FUNCTION_1ST_GEN;
+  }
+
+  if (
+    process.env.K_SERVICE !== undefined &&
+    process.env.FUNCTION_TARGET !== undefined
+  ) {
+    return CloudEnvironment.GOOGLE_CLOUD_RUN_FUNCTION_2ND_GEN;
+  }
+
+  return CloudEnvironment.UNKNOWN;
+}
+
+function getBinaryPath(): string {
+  if (process.env.DD_SERVERLESS_COMPAT_PATH !== undefined) {
+    return process.env.DD_SERVERLESS_COMPAT_PATH;
+  }
+
+  const binaryPathOsFolder =
+    process.platform === 'win32'
+      ? resolve(__dirname, '../bin/windows-amd64')
+      : resolve(__dirname, '../bin/linux-amd64');
```

Comment:
does hardcoding `/` here work on windows? i assume this should be

```suggestion
      ? resolve(__dirname, '..', 'bin', 'windows-amd64')
      : resolve(__dirname, '..', 'bin', 'linux-amd64');
```

Reply:
I hadn't run into any issues but I agree this change should be applied because windows shouldn't be using the forward slash `/`.